File: Selenium/Project/task6.java
package project;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import static org.testng.AssertJUnit.assertTrue;

public class task6 {
    WebDriver driver;
    // Logging into the site
    // Goal: Open the website and log-in using the provided credentials.

    @BeforeClass
    public void beforeMethod() {
        // Set up the firefox driver
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        driver = new FirefoxDriver();

        // Open a browser.
        // Navigate to ‘https://alchemy.hguy.co/lms’
        driver.get("https://alchemy.hguy.co/lms");
    }

    @Test
    public void loginButton() {
        // Find the navigation bar
        // Select the menu item that says “My Account” and click it
        WebElement navigationBar = driver.findElement(By.linkText("My Account"));
        String navigationBarText = navigationBar.getText();
        System.out.println("Navigation bar text is: " + navigationBarText);
        navigationBar.click();

        // Read the page title and verify that you are on the correct page
        String currentPageTitle = driver.getTitle();
        System.out.println("Current Page Title is: " + currentPageTitle);
        assertTrue(currentPageTitle.contains(navigationBarText));

        // Find the “Login” button on the page and click it
        driver.findElement(By.className("ld-login-text")).click();

        // Find the username field of the login form and enter the username into that field
        driver.findElement(By.xpath("//input[@id='user_login']")).sendKeys("root");

        // Find the password field of the login form and enter the password into that field
        driver.findElement(By.cssSelector("input[type='password']")).sendKeys("pa$$w0rd");

        // Find the login button and click it
        driver.findElement(By.xpath("//input[@value='Log In']")).click();

        // Verify that you have logged in
        WebElement verification = driver.findElement(By.linkText("Edit profile"));
        String verifyLogin = verification.getText();
        Assert.assertEquals(verifyLogin, "Edit profile");
        System.out.println("Verify Edit profile is coming: " + verifyLogin);
    }

    @AfterClass
    public void afterMethod() {
        // Close the browser
        driver.quit();
    }
}

File: Selenium/Project/task6.py
# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

# Logging into the site
# Goal: Open the website and log-in using the provided credentials


# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Navigate to ‘https://alchemy.hguy.co/lms’.
    driver.get("https://alchemy.hguy.co/lms/")

    # Find the navigation bar.
    # Select the menu item that says "My Account" and click it.
    navigationBar = driver.find_element(By.XPATH,"//*[@id='menu-item-1507']/a")
    navigationBarText = navigationBar.text
    navigationBar.click()
    print("Title of New webpage is:", navigationBarText)

    # Read the page title and verify that you are on the correct page
    currentPageTitle = driver.title
    print("Title of new page title is: ", currentPageTitle)
    assert navigationBarText in currentPageTitle

    # Find the “Login” button on the page and click it
    driver.find_element(By.CLASS_NAME,"ld-login-button").click()

    # Find the username field of the login form and enter the username into that field.
    # Find the password field of the login form and enter the password into that field.
    # Find the login button and click it.
    driver.find_element(By.CSS_SELECTOR,"input[id='user_login']").send_keys("root")
    driver.find_element(By.CSS_SELECTOR,"input[id='user_pass']").send_keys("pa$$w0rd")
    driver.find_element(By.XPATH,"//input[@type='submit']").click()

    # Verify that you have logged in
    Verification = driver.find_element(By.LINK_TEXT,"Edit profile").text
    print("Verification text is: ", Verification)
    assert "Edit profile" == Verification

    # If it matches, close the browser
    driver.quit